From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Softik Lord <dimap9986@gmail.com>
Date: Thu, 14 Jul 2022 04:57:46 +0500
Subject: [PATCH] perf: async spawn chunks


diff --git a/src/main/java/com/github/ruviolence/reaper/Constants.java b/src/main/java/com/github/ruviolence/reaper/Constants.java
new file mode 100644
index 0000000000000000000000000000000000000000..8e0e7542ed6d95dc9a403d320640651998abeb6e
--- /dev/null
+++ b/src/main/java/com/github/ruviolence/reaper/Constants.java
@@ -0,0 +1,5 @@
+package com.github.ruviolence.reaper;
+
+public class Constants {
+    public static final Runnable NULL_RUNNABLE = () -> {};
+}
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index cebb44c67a0116c6b281086fc84d7257028ece8e..3826c468f070b2131d4d8b3843327b17f243b2b7 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -397,7 +397,7 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
                     }
 
                     ++i;
-                    worldserver.getChunkProviderServer().getChunkAt(blockposition.getX() + k >> 4, blockposition.getZ() + l >> 4);
+                    worldserver.getChunkProviderServer().getChunkAt(blockposition.getX() + k >> 4, blockposition.getZ() + l >> 4, com.github.ruviolence.reaper.Constants.NULL_RUNNABLE); // Reaper - Async Spawn Chunks
                 }
             }
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index fe770f35b17ba685742f290266a9ef6086ee1148..c5e0a7272a54c7c4de1508747ac1648d6280d650 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1000,7 +1000,7 @@ public final class CraftServer implements Server {
                     }
 
                     BlockPosition chunkcoordinates = internal.getSpawn();
-                    internal.getChunkProviderServer().getChunkAt(chunkcoordinates.getX() + j >> 4, chunkcoordinates.getZ() + k >> 4);
+                    internal.getChunkProviderServer().getChunkAt(chunkcoordinates.getX() + j >> 4, chunkcoordinates.getZ() + k >> 4, com.github.ruviolence.reaper.Constants.NULL_RUNNABLE); // Reaper - Async Spawn Chunks
                 }
             }
         }
