From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ruViolence <78062896+ruViolence@users.noreply.github.com>
Date: Thu, 27 Apr 2023 04:15:54 +0800
Subject: [PATCH] perf: deactivate gameLoopFunction


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index ba3e4bf290c7308c8a961ca9232695f9b4051c37..263fd80949aa5aac573b56e0f1c4940f6c90f5ee 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1022,9 +1022,11 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
         this.v.tick();
         MinecraftTimings.playerListTimer.stopTiming(); // Spigot
         // this.methodProfiler.c("commandFunctions"); // Reaper - Remove MethodProfiler
+        if (false) { // Reaper start - Deactivate gameLoopFunction
         MinecraftTimings.commandFunctionsTimer.startTiming(); // Spigot
         this.aL().e();
         MinecraftTimings.commandFunctionsTimer.stopTiming(); // Spigot
+        } // Reaper end - Deactivate gameLoopFunction
         // this.methodProfiler.c("tickables"); // Reaper - Remove MethodProfiler
 
         MinecraftTimings.tickablesTimer.startTiming(); // Spigot
