From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <id_cn00@outlook.com>
Date: Fri, 5 Apr 2024 17:21:20 +0800
Subject: [PATCH] Fix-compatibility


diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
deleted file mode 100644
index 8596b4f2f68c87fa6f6fe0e78a0bb0231ed1fc1a..0000000000000000000000000000000000000000
--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+++ /dev/null
@@ -1,7 +0,0 @@
-package io.papermc.paper.configuration;
-
-public class GlobalConfiguration {
-    public class PacketLimiter {
-        // Beast - This is just a dummy class to prevent ViaVersion from enabling its packet limiter.
-    }
-}
