From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <id_cn00@outlook.com>
Date: Wed, 1 May 2024 00:33:54 +0800
Subject: [PATCH] Virtual-thread-for-async-executor


diff --git a/src/main/java/com/homomc/beast/BeastConfig.java b/src/main/java/com/homomc/beast/BeastConfig.java
index 52807051abe14e33e07b2cfcf0426927d4f798e7..386d6d2066ed9e0f56f9f149a4391eebc42a463e 100644
--- a/src/main/java/com/homomc/beast/BeastConfig.java
+++ b/src/main/java/com/homomc/beast/BeastConfig.java
@@ -156,6 +156,12 @@ public class BeastConfig {
         virtualThreadForScheduler = getBoolean("virtual-thread-for-scheduler", false);
     }
 
+    public static boolean virtualThreadForAsyncExecutor;
+
+    private static void virtualThreadForAsyncExecutor() {
+        virtualThreadForAsyncExecutor = getBoolean("virtual-thread-for-async-executor", false);
+    }
+
     public static int portalTravelCacheLife;
     public static boolean portalTravelInstantExpire;
     private static void portalTravel() {
diff --git a/src/main/java/net/minecraft/server/MCUtil.java b/src/main/java/net/minecraft/server/MCUtil.java
index 4539b56012ee2ffcfcce86212f609341e76a9887..70a1dbdb31d44fcabce54fd1b78f29b1848dcd26 100644
--- a/src/main/java/net/minecraft/server/MCUtil.java
+++ b/src/main/java/net/minecraft/server/MCUtil.java
@@ -3,6 +3,8 @@ package net.minecraft.server;
 import com.destroystokyo.paper.profile.CraftPlayerProfile;
 import com.destroystokyo.paper.profile.PlayerProfile;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.homomc.beast.BeastConfig;
+import com.homomc.beast.virtualthreads.VirtualThreadService;
 import com.mojang.authlib.GameProfile;
 import org.apache.commons.lang.exception.ExceptionUtils;
 import org.bukkit.Location;
@@ -12,14 +14,28 @@ import org.spigotmc.AsyncCatcher;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.lang.reflect.Method;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Executor;
 import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
 import java.util.function.Supplier;
 import java.util.regex.Pattern;
 
 public final class MCUtil {
-    private static final Executor asyncExecutor = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder().setNameFormat("Paper Async Task Handler Thread - %1$d").build());
+    private static Executor asyncExecutor;
+
+    static {
+        if (VirtualThreadService.getJavaMajorVersion() >= VirtualThreadService.minimumJavaMajorVersionWithoutFeaturePreview && BeastConfig.virtualThreadForAsyncExecutor) {
+            try {
+                Method newThreadPerTaskExecutor = Executors.class.getMethod("newThreadPerTaskExecutor", ThreadFactory.class);
+                asyncExecutor = (Executor) newThreadPerTaskExecutor.invoke(null, VirtualThreadService.get().createFactory());
+            } catch (Exception ignored) {
+                System.err.println("Failed to create Virtual Thread executor! Fallback to default executor.");
+            }
+        }
+        asyncExecutor = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder().setNameFormat("Paper Async Task Handler Thread - %1$d").build());
+    }
 
     private MCUtil() {}
 
